### DISCUSSION TITLE
Provide data for silence at beginning and end of tracks

### DISCUSSION THREAD

**Discussion by huantianad** [navidrome/navidrome#3281] [Ideas]
Previously, #2082 discussed the possibility of adding silence removal at the start and end of tracks, for better gapless playback. The conclusion was that this was better left to the clients. However, after discussion in the Feishin music player repo about this: https://github.com/jeffvli/feishin/issues/721, it seems that this is not possible for clients that stream the audio, as to properly remove silence at the end of tracks, we need to buffer the entire track.

A possible solution to this is if the server preprocesses the track beforehand, and is able to provide the exact lengths of silence, or the exact timestamp of when the actual track begins and ends, and sends this (somehow) to the client. Alternatively, we could go back to the idea of doing this entirely server-side.

**Comment by dweymouth** [comment]
If files that are supposed to play gaplessly instead have silence hard-coded at the end, that seems like an encoding problem, yes? Perhaps you need to seek out a better transcode of the files. Preferably using a format other than MP3 since MP3 doesn't always preserve exact file lengths

**Reply by huantianad** [reply]
Yeah, I think that makes sense for tracks that are meant to be played gaplessly. However, I think that there are other tracks that also benefit from gapless playback, even if not specifically designed for it, like a video game soundtrack, where each track has a fade-in/fade-out. In that case, I'd like to be able to preserve the original file and not have to modify it to remove the silence.
